### PR TITLE
[docs] Add how to SerDe for Java function

### PR DESCRIPTION
**Pull request by Jennifer88huang**
### Motivation

How to SerDe for Java Functions are not described in the Functions doc.

### Modifications

Add the description.

### CODE REVIEW

**Review by jiazhai — APPROVED** [review]

File: site2/docs/functions-develop.md [hunk 1]
```diff
@@ -104,22 +104,14 @@ For complete code, see [here](https://github.com/apache/pulsar/blob/master/pulsa
 <!--END_DOCUSAURUS_CODE_TABS-->
 
 ## Schema registry
-Pulsar has a built in schema registry and comes bundled with a variety of popular schema types(avro, json and protobuf). Pulsar Functions can leverage existing schema information from input topics and derive the input type. The schema registry applies for output topic as well.
+Pulsar has a built-in schema registry and is bundled with popular schema types, such as avro, json and protobuf. Pulsar Functions can leverage the existing schema information from input topics and derive the input type. The schema registry applies for output topic as well.
```
Comment (Huanli-Meng) [inline site2/docs/functions-develop.md:107]:
```suggestion
Pulsar has a built-in schema registry and is bundled with popular schema types, such as Avro, JSON and Protobuf. Pulsar Functions can leverage the existing schema information from input topics and derive the input type. The schema registry applies for output topic as well.
```

File: site2/docs/functions-develop.md [hunk 2]
```diff
@@ -129,6 +121,12 @@ public interface SerDe<T> {
     byte[] serialize(T input);
 }
 ```
+SerDe works in the following way in Java Functions.
```
Comment (Huanli-Meng) [inline site2/docs/functions-develop.md:124]:
```suggestion
SerDe works in the following ways in Java Functions.
```

File: site2/docs/functions-develop.md [hunk 3]
```diff
@@ -129,6 +121,12 @@ public interface SerDe<T> {
...
+- If the input and output topics have schema, Pulsar Functions use schema for SerDe.
+- If the input and output topics do not exist, Pulsar Functions adopt the following rules to determine SerDe:
+  - If the schema type is specified, Pulsar Functions use the specified schema type.
+  - If SerDe is specified, Pulsar Functions use the specified SerDe, and the schema type for input and output topics is `Byte`.
+  - If either the schema type nor SerDe is specified, Pulsar Functions use the built-in SerDe. For non-primitive schema type, the built-in SerDe serializes and deserializes objects in the `json` format.
```
Comment (Huanli-Meng) [inline site2/docs/functions-develop.md:129]: either...or, neither...nor, Please double confirm which one you should use.

File: site2/docs/functions-develop.md [hunk 4]
```diff
@@ -129,6 +121,12 @@ public interface SerDe<T> {
     byte[] serialize(T input);
 }
 ```
+SerDe works in the following way in Java Functions.
+- If the input and output topics have schema, Pulsar Functions use schema for SerDe.
+- If the input and output topics do not exist, Pulsar Functions adopt the following rules to determine SerDe:
```
Comment (Huanli-Meng) [inline site2/docs/functions-develop.md:126]: If there is no input and output topics, what the schema is used for? Do you mean that the input and output topics do not specify the schema?

**Review by Huanli-Meng — COMMENTED** [review]
leave my comments. PTAL

File: site2/docs/functions-develop.md (hunk 3 above)
Comment (Jennifer88huang) [reply site2/docs/functions-develop.md]: neither ... nor...
Nice catch, thank you.

**Review by wolfstudy — APPROVED** [review]
Cool, LGTM+1

File: site2/docs/functions-develop.md (hunk 4 above)
Comment (Jennifer88huang) [reply site2/docs/functions-develop.md:126]: In some scenarios, we do not have input or output topics. 
Maybe I should change "and" into "or".

**Pull request merged**